src/persist/src/azure.rs

```diff
@@ -216,35 +200,97 @@ impl Blob for AzureBlob {
             // buffer into lgalloc.
             let mut buffer = match content_length {
                 1.. => {
-                    let region = self
-                        .metrics
+                    let region = metrics
                         .lgbytes
                         .persist_azure
                         .new_region(usize::cast_from(content_length));
                     PreSizedBuffer::Sized(region)
                 }
-                0 => PreSizedBuffer::Unknown(Vec::new()),
+                0 => {
+                    metrics.get_invalid_resp.inc();
```

Comment:
The S3 metrics say that a "content-length of 0 isn't necessarily invalid", which makes sense to me. Could we inc this only if the size turns out to not match the header?

(Generally I'm not convinced of the need to have this defensive coding here, so it'd be handy if this metric fired only in the cases that it was actually load-bearing!)

Reply:
Ahhh yeah you're totally right, updated!